cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -604,6 +604,7 @@ public Void call() throws Exception {
     deleteAppMetadata(appId, appSpec);
 
     store.removeApplication(appId);
+    store.deleteWorkflowStats(appId);
```

Comment:
App deletion should happen last. After all the operational/metadata related is deleted.